src/styles/components/calendar.scss

```
@@ -0,0 +1,51 @@
+.calendar {
+  width: $calendar-width;
+  margin: 0 auto;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  transition: 0.2s ease-in-out;
+
+  // adding correct margin-left to needed ceil of day
+  @each $day, $dayNum in $daysWeek {
+    &--start-day-#{$day} {
+      .calendar__day:nth-child(1) {
+        margin-left: $ceil-width * $dayNum + $dayNum;
+      }
+    }
+  }
+
+  // disable ceils below count of day in month
+  @for $day from 1 through $daysCount {
+    &--month-length-#{$day} {
+      .calendar__day:nth-child(n + #{$day + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    @include square;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #fff;
+    font-size: 31px;
+    font-family: Arial, sans-serif;
+    transition: background-color, transform 0.5s ease-in-out;
```

```suggestion
    transition: background-color 0.5s ease-in-out, transform 0.5s ease-in-out;
```
In your case `background-color` doesn't change smoothly, you should specify the duration too